cmake/externals/json.cmake

```diff
@@ -1,11 +1,11 @@
-FetchContent_Declare(json
+FetchContent_Declare(nlohmann_json
   GIT_REPOSITORY https://github.com/nlohmann/json.git
   GIT_TAG v3.10.5)
```

Comment:
@snnn, @edgchen1 Did any of you know what happens if the version mismatch?

Reply:
The one who appears first will get used.